client/dashboard/sites/settings-site-visibility/privacy-form.tsx

```diff
@@ -140,29 +192,51 @@ export function PrivacyForm( {
 						<VStack spacing={ 4 }>
 							<DataForm< SiteSettings >
 								data={ formData }
-								fields={ fields }
-								form={ form }
-								onChange={ ( edits: Partial< SiteSettings > ) => {
-									setFormData( ( data ) => {
-										const newFormData = { ...data, ...edits };
-
-										if ( edits.wpcom_site_visibility !== undefined ) {
-											// Forget any previous edits to the discoverability controls when the visibility changes.
-											newFormData.wpcom_discourage_search_engines =
-												settings.wpcom_discourage_search_engines;
-											newFormData.wpcom_prevent_third_party_sharing =
-												settings.wpcom_discourage_search_engines ||
-												settings.wpcom_prevent_third_party_sharing;
-										}
-
-										if ( edits.wpcom_discourage_search_engines === true ) {
-											// Checking the search engine box forces the third party checkbox too.
-											newFormData.wpcom_prevent_third_party_sharing = true;
+								fields={ visibilityFields }
+								form={ visibilityForm }
+								onChange={ handleChange }
+							/>
+							{ formData.wpcom_site_visibility === 'public' && isPrimaryDomainStaging && (
+								<Notice
+									variant="warning"
+									density="medium"
+									actions={
+										hasNonWpcomDomain ? (
+											<Button variant="secondary" href={ `/domains/manage/${ site.slug }` }>
+												{ __( 'Manage domains' ) }
+											</Button>
+										) : (
+											<Button
+												variant="secondary"
+												href={ addQueryArgs( `/domains/add/${ site.slug }`, {
+													redirect_to: `/v2/sites/${ site.slug }/settings/site-visibility`,
```

Comment:
This won't work in the backported screens :') maybe we need to add this issue to DOTDASH-25 altogether

Reply:
Good catch! I believe the Calypso warning just uses `redirect_to: window.location` or something. So I should probably just do that.

Reply:
Are going to fix this as part of this PR?